File: 983-ValidateStackSequences/983-ValidateStackSequences.py
# Last updated: 8/13/2025, 8:18:11 PM
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        n = len(pushed)
        m = len(popped)
        stack = []
        if n!= m : 
            return False
        i = 0
        j = 0
        while i < n and j < m: 
            if len(stack) == 0 : 
                stack.append(pushed[i])
                i+=1
            
            else : 
                if popped[j] != stack[-1] : 
                    if i < n : 
                        stack.append(pushed[i])
                        i+=1
                    else : 
                        return False
                
                elif popped[j] == stack[-1] : 
                    stack.pop(-1)
                    j+=1
        
        if j < n : 
            while len(stack) > 0 and popped[j] == stack[-1] : 
                stack.pop(-1)
                j+=1
            if j < n and popped[j] != stack[-1] :
                return False
        
        return True

File: 983-ValidateStackSequences/983-ValidateStackSequences.java
// Last updated: 8/13/2025, 8:18:16 PM
class Solution {
    Stack<Integer> stack = new Stack();
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        int i = 0;
        int j= 0;
        while(i<pushed.length && j<popped.length){
            if(stack.size()==0){
                stack.push(pushed[i]);
                i++;
            }
            else{
                if(stack.peek()==popped[j]){
                    stack.pop();
                    j++;
                }
                else{
                    stack.push(pushed[i]);
                    i++;
                }
            }
        }
        if(i==pushed.length){
            while(j<popped.length){
                if(stack.peek()==popped[j]){
                    stack.pop();
                    j++;
                }
                else{
                    return false;
                }
            }
        }
        if(stack.size()==0){
            return true;
        }
        return false;
    }

}